src/tye_lab_to_nwb/ast_ophys/extractors/cnmfe_matlab_segmentationextractor.py

```diff
@@ -0,0 +1,72 @@
+from encodings.utf_8 import decode
+
+import numpy as np
+from pymatreader import read_mat
+from roiextractors import SegmentationExtractor
+from roiextractors.extraction_tools import PathType
+from scipy.sparse import csc_matrix
+
+
+class CnmfeMatlabSegmentationExtractor(SegmentationExtractor):
+    """A SegmentationExtractor for matlab CNMF-E segmentation output."""
+
+    extractor_name = "CnmfeMatlabSegmentation"
+    mode = "file"
+
+    def __init__(self, file_path: PathType):
+        """Initialize a CnmfeMatlabSegmentationExtractor instance.
+
+        Parameters
+        ----------
+        file_path: str
+            The location of the folder containing  CNMF-E .mat output file.
+        """
+        super().__init__()
+        self.file_path = file_path
+
+        self._dataset_file = self._load_mat_file()
+
+        assert "C" in self._dataset_file, (
+            f"The fluorescence traces cannot be extracted from {self.file_path}."
+            "Make sure to run `save_neuron.m` in MATLAB to load the file that was produced by CNMF_E "
+            "and save the contents of the 'neuron' struct in a new .mat file that can be read by Python."
+        )
+
+        self._sampling_frequency = float(self._dataset_file["Fs"])
+        # image axis order should be height and width
+        self._image_size = (self._dataset_file["options"]["d1"], self._dataset_file["options"]["d2"])
+        self._num_frames = self._dataset_file["P"]["numFrames"]
+        self._roi_response_raw = self._dataset_file["C"].T
+        self._roi_response_deconvolved = self._transform_deconvolved_traces()
+        self._image_correlation = self._dataset_file["Cn"]
+        self._image_masks = self._transform_image_masks()
```

Comment:
Did you get a chance to confirm all these mappings with @alessandratrapani?

Reply:
Yes! We discussed this today with @alessandratrapani and they should be correct.

Reply:
Awesome, thanks!